animation_nodes/nodes/vector/transform_vector.py

```diff
@@ -6,19 +6,27 @@ class TransformVectorNode(bpy.types.Node, AnimationNode):
     bl_label = "Transform Vector"
 
     useVectorList: VectorizedSocket.newProperty()
+    useMatrixList: VectorizedSocket.newProperty()
 
     def create(self):
         self.newInput(VectorizedSocket("Vector", "useVectorList",
             ("Vector", "vector"),
             ("Vectors", "vectors", dict(dataIsModified = True))))
+        self.newInput(VectorizedSocket("Matrix", "useMatrixList",
+            ("Matrix", "matrix"), ("Matrices", "matrices")))
 
-        self.newInput("Matrix", "Matrix", "matrix")
-
-        self.newOutput(VectorizedSocket("Vector", "useVectorList",
```

Comment:
VectorizedSocket can take multiple properties as input. `["useVectorList", "useMatrixList"]`. This should avoid the if statement.

Reply:
By using this method there is an issue with socket update.

https://user-images.githubusercontent.com/46858047/116970318-5a6c7180-acd5-11eb-96ec-9a123ef63421.mp4

Reply:
Seems like a standard behavior for the vectorization selector. Perhaps this behavior should change globally then.